platforms/software/signing/src/main/java/org/gradle/plugins/signing/Sign.java

```diff
@@ -322,6 +328,7 @@ protected FileCollectionFactory getFileCollectionFactory() {
      * All of the files that will be signed by this task.
      */
     @Internal
+    @ToBeKeptEagerProperty(because = "Read-only file collection")
```

Comment:
🤔 I wonder what we want to do with all these read-only `FileCollection`s in singing. I'd say we should deprecate them all...

Reply:
Added willBeDeprecated = true